src/gtk2.mk

```diff
@@ -19,7 +19,17 @@ define $(PKG)_UPDATE
     head -1
 endef
 
+define $(PKG)_PATCH_STATIC
+    (cd $(SOURCE_DIR) && $(PATCH) -p1 -u) < $(TOP_DIR)/src/gtk2-patch-static-dllmain.diff
+endef
+
+define $(PKG)_PATCH_X64
+    (cd $(SOURCE_DIR) && $(PATCH) -p1 -u) < $(TOP_DIR)/src/gtk2-remove-abi-compatibility-workarounds-x64.diff
+endef
+
 define $(PKG)_BUILD
+    $(if $(BUILD_STATIC), $(call $(PKG)_PATCH_STATIC))
+    $(if $(findstring x86_64, $(TARGET)), $(call $(PKG)_PATCH_X64))
```

Comment:
Variable patches for a single version are [deliberately excluded from the pkg rule lookup](https://github.com/mxe/mxe/blob/4ccfdfc560c63188c1c61a1abbb9e736cc985715/Makefile#L572). There's never a good reason for them:
  - can't be sent upstream
  - hard to maintain
  - don't fit with patch workflow
  - there's always a correct set of defines and macros to achieve the same result

The latter is, of course, a lot of work. It's better to leave features disabled till someone is prepared to do that work.

Reply:
> * there's always a correct set of defines and macros to achieve the same result
> 
> The latter is, of course, a lot of work. It's better to leave features disabled till someone is prepared to do that work.

Trust me, there isn't. I tried fixing them before I did this.
Several of the current patches in master I moved to the static patch because they were actively breaking the shared builds. By contrast, the x64 build waw always going to fail because of a lot of compatibility code thought out by them to play nice with older versions of tye the other library but we don't really care about that.

Keep in mind that this is gtk2 we're talking about. It's essentially an EoL package already.
The x64 patch was pulled from a commit under the `2.90.7` tag upstream.

Reply:
P.S. I went into the #gtk official channel at one point asking if they could help me solve some of the errors. Their literal reply? "Why do you still want gtk2 anyway?"